src/puppetlabs/trapperkeeper/services/webserver/jetty9_service.clj

```diff
@@ -26,23 +33,38 @@
 (defservice jetty9-service
   "Provides a Jetty 9 web server as a service"
   WebserverService
-  [[:ConfigService get-in-config]]
+  {:required [ConfigService]
+   :optional [FilesystemWatchService]}
   (init [this context]
         (log/info (i18n/trs "Initializing web server(s)."))
-        (let [config (or (get-in-config [:webserver])
+        (let [config-service (get-service this :ConfigService)
+              config (or (tk-config/get-in-config config-service [:webserver])
                          ;; Here for backward compatibility with existing projects
-                         (get-in-config [:jetty])
+                         (tk-config/get-in-config config-service [:jetty])
                          {})]
           (config/validate-config config)
           (core/init! context config)))
 
   (start [this context]
          (log/info (i18n/trs "Starting web server(s)."))
-         (let [config (or (get-in-config [:webserver])
+         (let [config-service (get-service this :ConfigService)
+               config (or (tk-config/get-in-config config-service [:webserver])
                           ;; Here for backward compatibility with existing projects
-                          (get-in-config [:jetty])
-                          {})]
-           (core/start! context config)))
+                          (tk-config/get-in-config config-service [:jetty])
+                          {})
+               started-context (core/start! context config)]
+           (if-let [filesystem-watcher-service
+                    (maybe-get-service this :FilesystemWatchService)]
+             (let [watcher (watch-protocol/create-watcher filesystem-watcher-service)]
+               (doseq [server (:jetty9-servers started-context)]
+                 (when-let [ssl-context-factory (-> server
+                                                    second
+                                                    :state
+                                                    deref
+                                                    :ssl-context-factory)]
+                   (core/reload-crl-on-change! ssl-context-factory watcher)))
+               (assoc started-context :watcher watcher))
+             started-context)))
```

Comment:
In the previous iteration it looks like we would always return the return value of `(core/start! context config)` which I think was "context". It looks like we've moved the call to `core/start!` inside the `let` definitions. Then within the `if-let` we return the context with the new file system watcher added. But if the `if-let` evaluates to false because we don't have a `FileSystemWatchService` it looks like the function would now return `nil`. Is that right, and if so ok?